MTKA-1310: querying taxonomies doc (#104)
wpengine/headless-docs: atlas-content-modeler/tutorial/models/taxonomies/query.mdx

---
slug: /atlas-content-modeler/tutorial/models/taxonomies/query
title: Atlas Content Modeler Taxonomies - Query

> description: Learn how to Query Taxonomies in Atlas Content Modeler.

Reviewer: ```suggestion
description: Learn how to query Models by Taxonomy in Atlas Content Modeler.
```

I think this more accurately describes the intent here?

---

**Note**: These tutorials assume that you have already [installed the plugin](../../../introduction.mdx).

## Query WPGraphQL for a Taxonomy

> Learn how to query taxonomies using WPGraphQL.

Reviewer: ```suggestion
Learn how to query models by taxonomy using WPGraphQL.
```
Missed this before, but I think it should read same/similar to the description above so that it clarifies you're querying/filtering content by taxonomy.

---

### Dependencies

#### WPGraphQL

Please download and install WPGraphQL following these instructions: [WPGraphQL Installation - Quick Start](https://www.wpgraphql.com/docs/quick-start/).

#### WPGraphQL Tax Query

> Please download, install, and activate **WPGraphQL Tax Query** using the [Wordpress Plugin Installation Method](https://wordpress.org/support/article/managing-plugins/#automatic-plugin-installation).

Reviewer: ```suggestion
Please download, install, and activate **WPGraphQL Tax Query** using the [WordPress Plugin Installation Method](https://wordpress.org/support/article/managing-plugins/#automatic-plugin-installation).
```

> Please download, install, and activate **WPGraphQL Tax Query** using the [Wordpress Plugin Installation Method](https://wordpress.org/support/article/managing-plugins/#automatic-plugin-installation).

Reviewer 2: Worth linking to the plugin here?

Also, I think people will have to install manually because WPGraphQL Tax Query isn't in the WordPress.org repo.

```suggestion
Please download, install, and activate [WPGraphQL Tax Query](https://github.com/wp-graphql/wp-graphql-tax-query) by [uploading the plugin zip file in the WordPress admin](https://wordpress.org/support/article/managing-plugins/#manual-upload-via-wordpress-admin).
```

Author: Updated

Once you have successfully installed and activated these plugins, you may continue with the steps below for Querying Taxonomies.

---

### Steps to Query a Taxonomy using WPGraphQL

You can find detailed steps in the basic usage of WPGraphQL here: [WPGraphQL Quick Start](https://www.wpgraphql.com/docs/quick-start/)

> The steps below are more specific on how to query content models by taxonomy. This assumes you have already [Created a Model](../create.mdx), and [Created a Taxonomy](create.mdx) for that model. The query below is in GraphiQL in the Wordpress plugin WPGraphQL.

Reviewer 2: ```suggestion
The steps below are more specific on how to query content models by taxonomy. This assumes you have already [Created a Model](../create.mdx), and [Created a Taxonomy](create.mdx) for that model. The query below is in GraphiQL in the WordPress plugin WPGraphQL.
```

---

#### Example Query 1

This example queries a Model called Employees. Employees uses the **LOCATION** Taxonomy. In this case, we are querying the term **dallas** and the taxonomy **LOCATION** inside of the **taxQuery: {}** object.

```

> {
>   employees(first: 10,
>     where: {
>       taxQuery: {
>         relation: OR,
>         taxArray: [
>           {
>             terms: ["dallas"],
>             taxonomy: LOCATION,
>             operator: IN,
>             field: SLUG
>           }
>         ]
>       }
>   	}
>   ){
>     nodes {
>       ...EmployeeFields
>     }
>   }
> }
> 
> fragment EmployeeFields on Employee {
>   name
>   employers {
>     edges {
>       node {
>         id
>       }
>     }
>   }
> }

Reviewer 2: @matthewguywright It might be worth simplifying this to remove the `relation: OR` (the query only uses one taxonomy so the relation isn't needed).

I also dropped the fragment to leave only `name` to make it a little easier to parse as an example, but see what you think. For me this query gives the same result as the original and works if the user doesn't have an 'employers' related field.

```suggestion
{
  employees(first: 10,
    where: {
      taxQuery: {        
        taxArray: [
          {
            terms: ["dallas"],
            taxonomy: LOCATION,
            operator: IN,
            field: SLUG
          }
        ]
      }
  	}
  ){
    nodes {
      name
    }
  }
}
```

Reviewer 2: Remove unused portions of the other example query.

```

#### Example Query 2

This example queries a Model called Companies. Companies uses the **REGION** Taxonomy. In this case, we are querying the term **east coast** and the taxonomy **REGION** inside of the **taxQuery: {}** object.

```

> {
>   companies(first: 2,
>     where: {
>       taxQuery: {
>         relation: OR,
>         taxArray: [
>           {
>             terms: ["east coast"],
>             taxonomy: REGION,
>             operator: IN,
>             field: SLUG
>           }
>         ]
>       }
>   	}
>   ){
>     nodes {
>       ...CompanyFields
>     }
>   }
> }
> 
> fragment CompanyFields on Company {
>   name
>   size
> }

Reviewer 2: Same here — we could simplify this query to remove unneeded parts if you want to:

```suggestion
{
  companies(first: 2,
    where: {
      taxQuery: {        
        taxArray: [
          {
            terms: ["east coast"],
            taxonomy: REGION,
            operator: IN,
            field: SLUG
          }
        ]
      }
  	}
  ){
    nodes {
      title
    }
  }
}
```
